Polyfills/URLSearchParams/Source/URLSearchParams.cpp

```diff
@@ -0,0 +1,165 @@
+    {
+        std::string queryStr = info[0].As<Napi::String>();
+        parseInputQueryStr(queryStr);
+    }
+
+    void URLSearchParams::parseKeyVal(const std::string& subStr)
+    {
+        auto equalSign = subStr.find("=");
+        if (equalSign != std::string::npos)
+        {
+            auto key = subStr.substr(0, equalSign);
+            auto value = subStr.substr(equalSign + 1);
+
+            params_vector.push_back(std::make_pair(key, value));
+            params_map[key] = value;
+        }
+    }
+
+    void URLSearchParams::parseInputQueryStr(std::string& queryStr)
+    {
+        // find all the &'s to determine where each param ends
+        // got positions for substring
+        if (queryStr.length() <= 1)
+        {
+            return;
+        }
+        
+        // if first character is ?, remove it
+        if (queryStr[0] == '?')
+        {
+            queryStr.erase(0, 1);
+        }
+        size_t start = 0;
+        size_t end = queryStr.find("&");
+
+        // find the first &, in a while loop
+        while (end != std::string::npos)
+        {
+            // get substring
+            std::string mySub = queryStr.substr(start, end - start);
+            parseKeyVal(mySub);
+
+            // update start, end
+            start = end + 1;
+            end = queryStr.find("&", start);
+        }
+
+        // now need to parse the last key val pair
+        parseKeyVal(queryStr.substr(start));
+    }
+
+    void URLSearchParams::setKeyVal(const std::string& key, const std::string& value) {
+        params_map[key] = value;
+        for (auto& paramsPair : params_vector) {
+            if (paramsPair.first == key) {
+                paramsPair.second = value;
+                return;
+            }
+        }
+        params_vector.push_back(std::make_pair(key, value));
+    }
+
+    Napi::Value URLSearchParams::Get(const Napi::CallbackInfo& info)
+    {
+        std::string key = info[0].As<Napi::String>();
+        std::string result;
+
+        auto element = params_map.find(key);
+
+        // element is not found 
+        if (element == params_map.end())
+        {
+            return Env().Null();
+        }
+        return Napi::Value::From(Env(), element->second);
+    }
+
+    std::string URLSearchParams::GetAllParams()
+    {
+        
+        std::string resultString = "";
+        if (params_vector.empty())
+        {
+            return resultString;
+        }
+
+        resultString += "?";
+
+        for (const auto& myPair : params_vector)
+        {
+            std::string key = myPair.first;
+            std::string value = myPair.second;
+
+            resultString += key;
+            resultString += "=";
+            resultString += value;
+            resultString += "&";
+        } 
+
+        // check last value of string 
+        char ch = resultString.back();
+        if (ch == '&')
+        { 
+            resultString.pop_back();
+        };
```

Comment:
I think it might be clearer if we append the first key/value outside the for loop and append the `&` first inside the first loop so that you don't need to do this odd check at the end. Or do it all in the for loop and check if it's the first element to know whether to append `?` or `&`.